bindings/java/src/main/com/apple/foundationdb/tuple/TupleUtil.java

```diff
@@ -63,6 +64,28 @@
 	private static final byte[] TRUE_ARR           = new byte[]{0x27};
 	private static final byte[] VERSIONSTAMP_ARR   = new byte[]{0x33};
 
+	private static final ThreadLocal<DecodeResult> DECODE_RESULT_THREAD_LOCAL =
```

Comment:
I have similar reservations about using a global (but thread local) decode result and encode result. Making decode result mutable seems sensible. Why not pass that around within the context of a decoding, and then this thread local constant can be removed?